src/coreclr/jit/llvmcodegen.cpp

```diff
@@ -325,10 +325,17 @@ void Llvm::generateBlocks()
         LlvmCompileDomTreeVisitor visitor(_compiler, this);
         visitor.WalkTree();
 
-        // Walk all the exceptional code blocks and generate them, since they don't appear in the normal flow graph.
+        // Walk all the exceptional code blocks and generate them since they don't appear in the normal flow graph.
         for (Compiler::AddCodeDsc* add = _compiler->fgGetAdditionalCodeDescriptors(); add != nullptr; add = add->acdNext)
         {
-            generateBlock(add->acdDstBlk);
+            // if the LLVM function was not created due to the first block not being reachable
+            // or the exceptional code block is part of a try that was not reachable,
+            // then don't generate the exceptional code block
+            if (m_functions[getLlvmFunctionIndexForBlock(add->acdDstBlk)].LlvmFunction != nullptr &&
+                !(add->acdDstBlk->hasTryIndex() && m_EHDispatchLlvmBlocks[add->acdDstBlk->getTryIndex()] == nullptr))
```

Comment:
Without this check it fails at 
```
 	clrjit_browser_wasm32_x64.dll!llvm::BasicBlock::getParent() Line 105	C++
	clrjit_browser_wasm32_x64.dll!Llvm::emitCallOrInvoke(llvm::FunctionCallee callee, llvm::ArrayRef<llvm::Value *> args) Line 2526	C++
 	clrjit_browser_wasm32_x64.dll!Llvm::buildCall(GenTreeCall * call) Line 1871	C++
 	clrjit_browser_wasm32_x64.dll!Llvm::visitNode(GenTree * node) Line 1088	C++
 	clrjit_browser_wasm32_x64.dll!Llvm::generateBlock(BasicBlock * block) Line 361	C++
 	clrjit_browser_wasm32_x64.dll!Llvm::generateBlocks() Line 339	C++
 	clrjit_browser_wasm32_x64.dll!Llvm::Compile() Line 29	C++
 	clrjit_browser_wasm32_x64.dll!Compiler::compCompile::__l2::<lambda_5>::operator()() Line 5182	C++
 	clrjit_browser_wasm32_x64.dll!ActionPhase<`Compiler::compCompile'::`2'::<lambda_5>>::DoPhase() Line 65	C++
```
because 
```cpp
catchLlvmBlock = m_EHDispatchLlvmBlocks[getCurrentProtectedRegionIndex()];
```
returns null.  However we can just put a `if (catchLlvmBlock != nullptr)` around the code that sets `catchLlvmBlock` to `nullptr` and it is fine with that.  I'll put that in the next commit

Reply:
Now removed the second condition and added a guard to the nulling of `catchLlvmBlock` in `emitCallOrInvoke`